docs-src/concepts/what-is-worker-versioning.md

```diff
@@ -0,0 +1,223 @@
+
+Accomplish this goal by assigning a Build ID (a free-form string) to the code that defines a Worker, and specifying which Build IDs are compatible with each other by updating the version sets associated with the Task Queue, stored by the Temporal Server.
+
+### When and why should you use Worker Versioning
+
+The main reason to use this feature is to deploy incompatible changes to short-lived [Workflows](/workflows). On Task Queues using this feature, the Workflow starter doesn't have to know about the introduction of new versions.
+
+The new code in the newly deployed Workers executes new [Workflow Executions](#workflow-execution), while only Workers with an appropriate version process old Workflow Executions.
+
+#### Decommision old Workers
+
+You can decommission old Workers once you archive all open Workflows using their version, or if you have no need to Query closed Workflows, you can decommission them once no open Workflows remain at that version.
+
+For example, if you have a Workflow that completes within a day. A good strategy is to assign a new Build ID to every new Worker build and add it as the new overall default in the version sets.
+
+Because your Workflow completes in a day, you know that you won't need to keep older Workers running for more than a day after you deploy the new version (assuming availability).
+
+Removing old versions from the sets isn't necessary. Leaving them doesn't cause any harm.
+
+You can apply this technique to longer-lived Workflows too; however, you may need to run multiple Worker versions simultaneously while open Workflows complete.
+
+#### Deploy code changes to Workers
+
+The feature also allows you to implement compatible changes to or prevent a buggy code path from executing on currently open Workflows. You can achieve this by adding a new version to an existing set and defining it as _compatible_ with an existing version, which shouldn't execute any future Workflow Tasks. Since the new version processes existing [Event Histories](/workflows/#event-history), it must adhere to the usual [deterministic constraints](/workflows/#deterministic-constraints), and you may need to use one of the [versioning APIs](/workflows/#workflow-versioning).
+
+Moreover, this feature enables you to make incompatible changes to Activity Definitions in conjunction with incompatible changes to Workflow Definitions that use those Activities. This functionality works because any Activity that a Workflow schedules on the same Task Queue only gets dispatched to Workers compatible with the Workflow that scheduled it.
+If you want to change an Activity Definition's type signature while creating a new incompatible Build ID for a Worker, you can do so without worrying about the Activity failing to execute on some other Worker with an incompatible definition. The same principle applies to Child Workflows.
```

Comment:
```suggestion
Moreover, this feature let you make incompatible changes to Activity Definitions in conjunction with incompatible changes to Workflow Definitions that use those Activities.
This functionality works because any Activity that a Workflow schedules on the same Task Queue gets dispatched only to Workers compatible with the Workflow that scheduled it.
If you want to change an Activity Definition's type signature while creating a new incompatible Build ID for a Worker, you can do so without worrying about the Activity failing to execute on some other Worker with an incompatible definition.
The same principle applies to Child Workflows.
```